woocommerce/woocommerce
# Fix `Add To Cart form` layout in the post editor

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





The Gutenberg editor within the Post Editor page is not encapsulated within an iframe. As a result, the global WordPress styles are being applied to the blocks. This leads to the generic WordPress CSS having a higher priority over the Add To Cart form layout styles.

To resolve this issue, the conflicting generic WordPress styles are overridden by directly applying the necessary styles to the Add To Cart form using the style attribute.


| Before | After |
|--------|--------|
| ![image](https://github.com/user-attachments/assets/6f68568b-57d8-41dd-a8dd-f9c469d11d63) | ![image](https://github.com/user-attachments/assets/85128f8a-2803-48a6-8a72-6e0ae0716424) | 


Closes #45369 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. On a new page, add `Single Product` block.
2. Select any product.
3. Ensure that the `Add To Cart Form` looks like the `After` column.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

Fix Add To Cart form layout in the post editor.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
